libcodechecker/cmd/cmd_line_client.py

```diff
@@ -365,17 +410,26 @@ def report_to_html(client, reports, output_dir):
 
         file_report_map = defaultdict(list)
         for report in reports:
-            file_report_map[report.fileId].append(report)
+            filePath = ""
+            if isinstance(report, Report):
+                filePath = report.main['location']['file_name']
+            else:
+                filePath = report.checked_file
+            file_report_map[filePath].append(report)
 
         file_cache = {}
-        for file_id, file_reports in file_report_map.items():
-            checked_file = file_reports[0].checkedFile
+        for file_path, file_reports in file_report_map.items():
+            checked_file = file_path
             filename = os.path.basename(checked_file)
+            h = int(hashlib.md5(file_path).hexdigest(), 16) % (10 ** 8)
```

Comment:
Why do we convert the hash to a number and limit the values to 10^8? This way the probability of hash collision is a much larger risk. Would not be better to use the full hash?

Reply:
Hash collision is minimal, as a file name is always the prefix. So this number is only to distinguish between files with the same name and different paths.
The full hash is 16 bytes long and in hex it is a 32 long character chain.
I think a 8 long decimal chain is enough to distinguish between files.
So an output HTML will look like:
alma.c_95479304.html

instead of 
alma.c_ebbc3c26a34b609dc46f5c3378f96e08.html


See

>>> h.digest_size
16L
>>> h.hexdigest()
'ebbc3c26a34b609dc46f5c3378f96e08'
>>> int(h.hexdigest(), 16) % (10 ** 8)
95479304L
>>>